ILRepack/ILRepack.cs

```diff
@@ -350,11 +359,24 @@ public void Repack()
                     Directory.CreateDirectory(outputDir);
                 }
 
+                Logger.Info("Writing output assembly to disk");
                 TargetAssemblyDefinition.Write(Options.OutputFile, parameters);
-
+                
                 sourceServerDataStep.Write();
 
-                Logger.Info("Writing output assembly to disk");
+                foreach (var assembly in MergedAssemblies)
+                {
+                    assembly.Dispose();
+                }
+                
+                TargetAssemblyDefinition.Dispose();
+                GlobalAssemblyResolver.Dispose();
+
+                win32ResourceStep.Patch(Options.OutputFile);
```

Comment:
Change of assembly after it was produces will make strong name signature invalid. All changes should be done in cecil object model before we call `TargetAssemblyDefinition.Write`. As workaround, loading patched assembly and writing it with cecil again may fix it.

Reply:
I temporary reverting to forked cecil (but upgraded to latest one: b69b6de92fd157bd7887846a79cbef1f94d78772). I thinnk the proper way to solve it will be creating PR to cecil repo (gluck/cecil@d9b47f346e061c0f0e9e6e431262533955a6caed - see discussion #79).